src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.Block.cs

```diff
@@ -1043,6 +1043,12 @@ private void EndTagBlock(Stack<Tuple<HtmlSymbol, SourceLocation>> tags, bool com
             if (!Context.DesignTimeMode)
             {
                 AcceptWhile(HtmlSymbolType.WhiteSpace);
+                if (Context.LastSpan.Kind == SpanKind.Transition)
+                {
+                    // Mark the whitespace at the end of a <text> tag as code.
+                    Span.CodeGenerator = new StatementCodeGenerator();
+                    Output(SpanKind.Code);
```

Comment:
Dangerous to just `Output(SpanKind.Code)` blindly here. Your goal is to only output the whitespace but another `Output` has not occurred prior to this code execution within this method.